// Copyright 2020 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator_options.proto";

message LandmarksSmoothingCalculatorOptions {
  extend CalculatorOptions {
    optional LandmarksSmoothingCalculatorOptions ext = 325671429;
  }

  // Default behaviour and fast way to disable smoothing.
  message NoFilter {}

  message VelocityFilter {
    // Number of value changes to keep over time.
    // Higher value adds to lag and to stability.
    optional int32 window_size = 1 [default = 5];

    // Scale to apply to the velocity calculated over the given window. With
    // higher velocity `low pass filter` weights new values higher.
    // Lower value adds to lag and to stability.
    optional float velocity_scale = 2 [default = 10.0];

    // If calculated object scale is less than given value smoothing will be
    // disabled and landmarks will be returned as is.
    optional float min_allowed_object_scale = 3 [default = 1e-6];
  }

  oneof filter_options {
    NoFilter no_filter = 1;
    VelocityFilter velocity_filter = 2;
  }
}
